### PR TITLE
Add NTP status message

### PR DESCRIPTION
**Pull request by EliRoberts5**
When the network time is updated via Button 2 , a message is displayed while the update takes place. An NTP status method is added. Some minor BLE changes are added

### CODE REVIEW

File: include/locales/en-US.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #define LANG_MANUALLY "Manually"
 #define LANG_CANCEL "Cancel"
 #define LANG_SAVE "Save"
+#define LANG_NTP_QUEUED "NTP Updating"
```
Comment (RuffaloLavoisier) [inline include/locales/en-US.h:14]: Setup is required for all languages that currently exist.

File: include/services/OswServiceTaskWiFi.h [hunk 2]
```diff
@@ -44,6 +44,8 @@ class OswServiceTaskWiFi : public OswServiceTask {
     const String& getStationPassword() const;
 
     ~OswServiceTaskWiFi() {};
+     // ** updated added NTP status method to return NTP status 
+    bool getNTPstatus(void);
```
Comment (RuffaloLavoisier) [inline include/services/OswServiceTaskWiFi.h:48]: Just comment `// blah blah`.
And `bool getNTPstatus();`

File: src/services/OswServiceTaskWiFi.cpp [hunk 3]
```diff
@@ -209,6 +209,13 @@ bool OswServiceTaskWiFi::isConnected() {
     return this->m_enableWiFi and (this->m_enableStation or WiFi.status() == WL_CONNECTED);
 }
 
+// ** updated NTP status thing paul october 07 2022
+bool OswServiceTaskWiFi::getNTPstatus(void){
+    
+    return this->m_waitingForNTPUpdate;
```
Comment (RuffaloLavoisier) [inline src/services/OswServiceTaskWiFi.cpp:215]: @simonmicro 
I think we need an update on this.

File: src/apps/tools/OswAppTimeConfig.cpp [hunk 4]
```diff
@@ -180,9 +180,21 @@ void OswAppTimeConfig::loop() {
         if (OswServiceAllTasks::wifi.isConnected()) {
             OswUI::getInstance()->setTextCursor(BUTTON_2);
             hal->gfx()->print(LANG_TFW_UPDATE);
+            // ** if time is still being updated reprint this status on the display
+            if (OswServiceAllTasks::wifi.getNTPstatus()  ) {    
+                hal->gfx()->setTextCursor(210,150);
+                hal->gfx()->print(LANG_NTP_QUEUED);  // ** NTP Updating message
+            }
```
Comment (RuffaloLavoisier) [inline src/apps/tools/OswAppTimeConfig.cpp:187]: Maybe
```
// if time is still being updated reprint this status on the display
if (OswServiceAllTasks::wifi.getNTPstatus()  ) {    
    hal->gfx()->print(LANG_NTP_QUEUED);  // ** NTP Updating message
} else {
    hal->gfx()->print(LANG_TFW_UPDATE);
}
```

File: src/apps/tools/OswAppTimeConfig.cpp [hunk 5]
```diff
@@ -180,9 +180,21 @@ void OswAppTimeConfig::loop() {
...
             if (hal->btnHasGoneDown(BUTTON_2)) {
                 if (OswServiceAllTasks::wifi.isConnected()) {
                     OswServiceAllTasks::wifi.queueTimeUpdateViaNTP();
+                    // ** when time update button2 is pressed, display NTP updating message
+                    // ** until update is complete
+                    hal->gfx()->setTextCursor(210,150);
+                    hal->gfx()->print(LANG_NTP_QUEUED);  // ** NTP Updating message
+                    #ifndef NDEBUG
+                        Serial.println("updating time via NTP");
+                    #endif
```
Comment (RuffaloLavoisier) [inline src/apps/tools/OswAppTimeConfig.cpp:197]: Please remove it.

**Review by RuffaloLavoisier — CHANGES_REQUESTED** [review]
Thank you for your contribution.

We are currently developing the BLE part, so please exclude this part separately.
So please remove `BLE` and `PIO` config file from git history?

@simonmicro 
Is it worth merging only NTP update notifications into `bugs/tooAggressiveTimeSync` branch?

File: include/locales/en-US.h (hunk 1 above)
Comment (EliRoberts5) [reply include/locales/en-US.h:14]: ok, unless there is another way to do this, I would be utilizing Google translate for all languages other then english and french.

File: include/services/OswServiceTaskWiFi.h (hunk 2 above)
Comment (EliRoberts5) [reply include/services/OswServiceTaskWiFi.h:48]: can you be more specific, do you request that the comment be removed?

Comment (RuffaloLavoisier) [reply include/services/OswServiceTaskWiFi.h:48]: What is that `**`?  ..  😄

File: src/apps/tools/OswAppTimeConfig.cpp (hunk 4 above)
Comment (EliRoberts5) [reply src/apps/tools/OswAppTimeConfig.cpp:187]: ok, I will look at it ..

File: include/services/OswServiceTaskWiFi.h (hunk 2 above)
Comment (RuffaloLavoisier) [reply include/services/OswServiceTaskWiFi.h:48]: And Just `bool getNTPstatus();` not `bool getNTPstatus(void);`.
For the other things and the rules.

Comment (EliRoberts5) [reply include/services/OswServiceTaskWiFi.h:48]: ok you want the "void" removed ..

File: src/apps/tools/OswAppTimeConfig.cpp (hunk 5 above)
Comment (EliRoberts5) [reply src/apps/tools/OswAppTimeConfig.cpp:197]: you want the message removed?

Comment (RuffaloLavoisier) [reply src/apps/tools/OswAppTimeConfig.cpp:197]: remove code.

Comment (EliRoberts5) [reply src/apps/tools/OswAppTimeConfig.cpp:197]: ok, it should still work without that code, I will have to test it to see, then I will remove it if that is the case. good call I think, although the user will not get any indication when they actually press the button, only when the update is still in progress. If you prefer this, I would appreciate if you would confirm.

Comment (EliRoberts5) [reply src/apps/tools/OswAppTimeConfig.cpp:197]: I also have a fairly large update for the watchfaces, wondering if you have any comment on how to submit those changes?

**Comment by EliRoberts5** [comment]
I am going to redo this and resubmit as a new PR.

**Pull request closed without merging**